opensearch-operator/controllers/opensearchController.go

```diff
@@ -149,17 +147,53 @@ func (r *OpenSearchClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 }
 
 // delete associated cluster resources //
-func (r *OpenSearchClusterReconciler) deleteExternalResources(cluster *opsterv1.OpenSearchCluster) error {
-	namespace := cluster.Spec.General.ClusterName
-	nsToDel := builders.NewNsForCR(cluster)
+func (r *OpenSearchClusterReconciler) deleteExternalResources(ctx context.Context) (ctrl.Result, error) {
```

Comment:
Rather than explicitly deleting resources it might be cleaner to add owner metadata to the resources and allow Kubernetes garbage collection to clean up. The controller-manager package has a built in way to add these references e.g.:
```go
err := ctrl.SetControllerReference(r.opensearchCluster, resource, r.client.Scheme())
```
Ideally we should only be using finalizers and deletion methods where there are resources external to the cluster to clean up.

Reply:
Good idea. I changed the code to use owner references for everything. I left the `DeleteResources` methods in, even if they are empty currently just so we have the mechanism in place when we need it.
One note: With envtest the garbage collection does not happen and objects do not get deleted (see https://book.kubebuilder.io/reference/envtest.html#testing-considerations), as such I changed the relevant tests to check for existence of an owner reference instead.